articles/grand-rename.md

```diff
@@ -0,0 +1,97 @@
+
+Our main goals with our new naming scheme were to be internally consistent, but at the same time make sure the names are abstract enough to work flexibly for changes to come in the foreseeable future. Too abstract names can be confusing and cumbersome, so we additionally strived for clarity.
+
+The most obvious change was to rename our Swift repository from "SwiftClient" to "SwiftSDK". Thanks to GitHub redirections, this change should have no effect on our existing customers, but long term, it should make things more consistent. 🎉
+
+For our signals & parameters, we decided the prefix them with `TelemetryDeck` to make the source clear and avoid amgiguity. Additionally, we analyzed all the signals and parameters we currently send and future ones we have in our roadmap and grouped them as follows:
+
+### Signals
+
+- `Session`: Anything related to starting, pausing, continuing, or ending a user session.
+
+### Parameters
+
+- `AppContext`: The context the app runs in, such as the locale, the debug mode, or the target environment.
+- `AppInfo`: Information about the specific app build, such as version, build number, or SDKs compiled with.
+- `Device`: All about the device running the application, such as operating system, model name, or architecture.
+- `SDK`: Information about the TelemetryDeck SDK, such as the version number or supported features.
+
+{% noteinfo "No action needed" %}
+To make sure these changes don't affect existing customers with existing insights anytime soon, we decided to keep sending the old names for a transition period of at least 1 year while also sending the new ones. We have also auto-migrated any insights that were using the old names in their filters to work with the new system. This was safely achieved by replacing filters like "platform == iOS" with something like "platform == iOS OR TelemetryDeck.SystemInfo.platform == iOS" accepting both styles, ensuring you will continue to see historic data from before the transition, during the transition, and after the transition.
+{% endnoteinfo %}
+
+<!-- Discussion: What about Top-N insights that use a parameter like "appVersion" which got renamed to "TelemetryDeck.SystemInfo.appVersion"? -->
+
+## Filling the Gaps
+
+While doing the rename, we also noticed some missing variants of parameters that could be useful. So we added the following variants:
+
+- `TelemetryDeck.AppContext.region`: The regional identifier of the users country setting, e.g. `US` for United States. Used for things like date format.
+- `TelemetryDeck.AppContext.language`: The language the app is currently used in (one of the supported languages).
+- `TelemetryDeck.AppInfo.versionAndBuildNumber`: Combines the app version and build into one String, such as `1.7.1 (build 22)`.
+- `TelemetryDeck.Device.preferredLanguage`: The language most preferred by the user on this device (might not be supported by app).
+- `TelemetryDeck.Device.type`: One of `Phone`, `Tablet`, `Desktop`, `Watch`, `TV`, or `Headset`
```

Comment:
We don't need that information in the SDK, because it can be derived from the modelName. To do that, we use Druid's lookups, and we feed them with [this data](https://github.com/TelemetryDeck/AppleModelNames).

Reply:
Where do you use Druid's lookups? Can users easily use the device type as well? I can't find any field here:
<img width="531" alt="Bildschirmfoto 2024-04-22 um 17 16 19" src="https://github.com/TelemetryDeck/docs/assets/6942160/8904fb16-e1fe-4f24-a207-4b8127d8043a">

Even if that was possible, I feel like doing such a device type lookup afterwards seems overkill to me and requires regular updates of the library. At least on all Apple platforms, getting the device type is super easy so the SDK could do it. Haven't checked for Android though, maybe it's harder there...

Reply:
Touché, lookups can't be set via the visual editor yet. But I think that should be an easy addition.